tests/integration/deploy_test.go

```diff
@@ -0,0 +1,418 @@
+package integration
+
+import (
+	"github.com/Optum/dce-cli/cmd"
+	"github.com/Optum/dce-cli/configs"
+	"github.com/Optum/dce-cli/internal/constants"
+	"github.com/Optum/dce-cli/internal/util"
+	"github.com/Optum/dce-cli/mocks"
+	"github.com/Optum/dce-cli/pkg/service"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestSystemDeployCommand(t *testing.T) {
```

Comment:
I wanted some tests against the deploy command with as little mocking as possible, so we can better simulate real-world behavior.

These tests mock services at the "edges" of the system -- Github downloads, AWS S3 uploads, and Terraform applies. But they execute the actual cobra command with CLI arguments, env vars, and YAML config files.